app/models/course/lecture.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+class Course::Lecture < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  has_many_attachments on: :content
+
+  belongs_to :course, inverse_of: :lectures
+
+  scope :sorted_by_date, -> { order(start_at: :desc) }
+  scope :sorted_by_sticky, -> { order(sticky: :desc) }
+
+  def to_partial_path
+    'course/announcements/announcement'
+  end
+
+  TIME_BRAINCERT = '%I:%M%p'
+  DATE_ISO = '%Y-%m-%d'
+  BRAINCERT_API_BASE_URL = 'https://api.braincert.com'
+
+  def handle_access_link(user, is_instructor)
+    if (msg = lecture_inactive?).present?
+      return nil, msg
+    end
+    create_classroom unless classroom_id
+    generate_classroom_link user, is_instructor
+  end
+
+  private
+
+  def generate_classroom_link(user, is_instructor)
+    res = call_braincert_api '/v2/getclasslaunch',
```

Comment:
Use `call_braincert_api()` with brackets, our convention only uses DSL form if we don't care about the return value.